internal/core/data/event.go

```diff
@@ -236,273 +208,113 @@ func updateEventPushDate(id string) error {
 	return nil
 }
 
-// Get event by device id
-// Returns the events for the given device sorted by creation date and limited by 'limit'
-// {deviceId} - the device that the events are for
-// {limit} - the limit of events
-// api/v1/event/device/{deviceId}/{limit}
-func getEventByDeviceHandler(w http.ResponseWriter, r *http.Request) {
-	defer r.Body.Close()
-
-	vars := mux.Vars(r)
-	limit := vars["limit"]
-	deviceId, err := url.QueryUnescape(vars["deviceId"])
-
-	// Problems unescaping URL
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		LoggingClient.Error("Error unescaping URL: " + err.Error())
-		return
-	}
-
-	// Convert limit to int
-	limitNum, err := strconv.Atoi(limit)
+// Put event on the message queue to be processed by the rules engine
+func putEventOnQueue(e models.Event) {
+	LoggingClient.Info("Putting event on message queue", "")
+	//	Have multiple implementations (start with ZeroMQ)
+	err := ep.SendEventMessage(e)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		LoggingClient.Error("Error converting to integer: " + err.Error())
-		return
+		LoggingClient.Error("Unable to send message for event: " + e.String())
 	}
+}
 
-	// Check device
-	if checkDevice(deviceId, w) == false {
-		return
+func testLimit (limit int) error {
```

Comment:
Since "test" has a specific connotation in Go, I think I'd recommend calling this "checkMaxLimit"